### PR TITLE
450 availability tests

### PR DESCRIPTION
**Pull request by banukutlu**
Issue #450 

Additional work: Updated sample data and fixtures with the latest.

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -9,20 +9,26 @@
...
 # Configuration parameters: CountComments.
 Metrics/ClassLength:
   Max: 227
 
+# Offense count: 8
+# Configuration parameters: Max.
+RSpec/ExampleLength:
+  Exclude:
+    - 'spec/features/availability_spec.rb'
+
```
Comment (cdmo) [inline .rubocop_todo.yml]: I just added a exclusion on ExampleLength for all features. Cool with that?

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -9,20 +9,26 @@
...
   Exclude:
     - 'spec/routing/**/*'
     - 'spec/controllers/catalog_controller_spec.rb'
 
-# Offense count: 9
+# Offense count: 16
 # Configuration parameters: AggregateFailuresByDefault.
 RSpec/MultipleExpectations:
-  Max: 3
+  Max: 6
```
Comment (cdmo) [inline .rubocop_todo.yml]: I just added a exclusion on MultipleExpectations for all features. Cool with that?

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -9,20 +9,26 @@
 # Offense count: 1
 # Configuration parameters: CountComments, ExcludedMethods.
 Metrics/BlockLength:
-  Max: 42
+  Max: 28
```
Comment (cdmo) [inline .rubocop_todo.yml]: Wonder why this changed. Max block length is now 28? 🤔

File: .rubocop_todo.yml (hunk 2 above)
Comment (banukutlu) [reply .rubocop_todo.yml]: 👍

File: .rubocop_todo.yml (hunk 1 above)
Comment (banukutlu) [reply .rubocop_todo.yml]: 👍

File: .rubocop_todo.yml (hunk 3 above)
Comment (banukutlu) [reply .rubocop_todo.yml]: could it be the changes related to availability maybe 🤔

**Comment by banukutlu** [comment]
> Could you rebase and force push? Just to resolve the conflicts.

resolved conflicts 👍

File: .rubocop_todo.yml (hunk 2 above)
Comment (cdmo) [reply .rubocop_todo.yml]: so how will we resolve this? I'll rebase my PR when this goes in to master and overwrite it? or if my PR goes in first we can rebase this branch? the race is on! ;)

Comment (banukutlu) [reply .rubocop_todo.yml]: I rebased this branch 👍

File: spec/features/availability_spec.rb [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+require 'rails_helper'
+
+RSpec.feature 'Availability', type: :feature do
+  describe 'User searches for a record', js: true do
+    it 'that has holdings and a \'View Availability\' button' do
+      visit '/?utf8=✓&search_field=all_fields&q=9781599901091'
+      expect(page).to have_selector 'button[data-target="#availability-5112336"]'
+    end
+
+    it 'that has holdings but no availability snippet' do
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
    it 'that has holdings but none are available' do
```

File: spec/features/availability_spec.rb [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      visit '/?utf8=✓&search_field=all_fields&q=9781599901091'
+      expect(page).not_to have_selector '.availability-snippet'
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
      expect(page).not_to have_selector '.availability-snippet'  # Shouldn't show the location teaser text next to the show availability button
```

File: spec/features/availability_spec.rb [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    end
+
+    it 'that has holdings and an availability snippet for multiple locations' do
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
    it 'that has holdings available in 3 or more locations' do
```

File: spec/features/availability_spec.rb [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+                                    exact_text: 'Multiple Locations'
+    end
+
+    it 'that has holdings and an availability snippet with 2 locations' do
+      visit '/?utf8=✓&search_field=all_fields&q=0060125896'
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Altoona, Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'that has holdings and an availability snippet with one location' do
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
    it 'that has holdings available in only 1 location' do
```

File: spec/features/availability_spec.rb [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+      expect(page).not_to have_selector '.availability-snippet'
+    end
+
+    it 'that has holdings and an availability snippet for multiple locations' do
+      visit '/?utf8=✓&search_field=all_fields&q=0802132138'
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Multiple Locations'
+    end
+
+    it 'that has holdings and an availability snippet with 2 locations' do
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
    it 'that has holdings available in exactly 2 locations' do
```

File: spec/features/availability_spec.rb [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+                                    exact_text: 'Altoona, Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'that has holdings and an availability snippet with one location' do
+      visit '/?utf8=✓&search_field=all_fields&q=9780147513861'
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'uses \'View Availability\' button to display and hide holdings' do
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
    it 'and clicks the \'View Availability\' button to display and hide holdings' do
```

File: spec/features/availability_spec.rb [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+  describe 'User opens a record with many holdings in a library' do
+    it 'toggles 4 or more holdings with View More/View Less button', js: true do
+      visit '/catalog/1793712'
+      within 'div[data-library="UP-MICRO"]' do
+        expect(page).to have_selector 'button', text: /View More/
+        expect(page).not_to have_selector 'button', text: /View Less/
+        expect(page).to have_xpath './/tbody/tr', count: 4
+        click_button('View More')
+        sleep 1 # let collapse animation finish and wait for it to re-collapse
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
```

File: spec/features/availability_spec.rb [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+      expect(page).to have_selector '.availability-snippet',
+                                    exact_text: 'Pattee Library and Paterno Library Stacks'
+    end
+
+    it 'uses \'View Availability\' button to display and hide holdings' do
+      visit '/?utf8=✓&search_field=all_fields&q=9781599901091'
+      expect(page).to have_selector 'button[data-target="#availability-5112336"]'
+      expect(page).not_to have_selector '.availability-5112336'
+      click_button('View Availability')
+      sleep 1 # let collapse animation finish and wait for it to re-collapse
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
```

`sleep` can be tricky

File: spec/features/availability_spec.rb [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+      expect(page).not_to have_selector '.availability-5112336'
```
Comment (cdmo) [inline spec/features/availability_spec.rb]:
```suggestion
      expect(page).not_to have_selector '.availability-5112336', wait: 3
```

File: spec/features/availability_spec.rb [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+        click_button('View Less')
+        expect(page).to have_xpath './/tbody/tr', count: 4
+      end
+    end
+  end
+
+  describe 'Archival Material:' do
+    it 'has \'Request Material\' link so it can be requested through Aeon', js: true do
+      visit '/catalog/1836205'
+      expect(page).to have_css 'a[data-type="aeon-link"][data-call-number="RBM 2457 box1 AX/SP/10155/11"]'
```
Comment (cdmo) [inline spec/features/availability_spec.rb]: This failed occasionally for me locally, but I'm not sure anything can be done about this without significant effort.

**Pull request merged**